src/test/resources/features/AddProductToCart.feature

```diff
@@ -3,8 +3,45 @@ Feature: Add product to Shopping cart
 
   Background:
     Given user is on Home page
+    And user clicks on 'electronics' category from category bar
+    And user clicks on 'cell-phones' subcategory
 
   Scenario: Validate access to subcategory
-    When User clicks on 'electronics' category from category bar
-    And User clicks on 'cell-phones' subcategory
-    Then 'Cell phones' subcategory page is displayed
\ No newline at end of file
+    Then 'Cell phones' subcategory is displayed
+
+  Scenario: Add product to the Shopping cart
+    When user clicks on 'htc-one-mini-blue' product
```

Comment:
And the next 3 steps also should be moved to the background, after all permutations 4 scenarios should remain so you should add one more scenario

Reply:
I want to upgrade and maybe change my feature file when we get a final decision with methods.
Thank You!